simple-generated/src/event.rs

```
@@ -1,87 +1,116 @@
-                            CrosstermEvent::FocusLost => Ok(()),
-                            CrosstermEvent::Paste(_) => unimplemented!(),
-                        }
-                        .expect("failed to send terminal event")
-                    }
-
-                    if last_tick.elapsed() >= tick_rate {
-                        sender.send(Event::Tick).expect("failed to send tick event");
-                        last_tick = Instant::now();
-                    }
-                }
-            })
-        };
-        Self {
-            sender,
-            receiver,
-            handler,
-        }
+        thread::spawn(move || event_thread(sender, tick_rate));
+        Self { receiver }
     }
 
     /// Receive the next event from the handler thread.
     ///
-    /// This function will always block the current thread if
-    /// there is no data available and it's possible for more data to be sent.
-    pub fn next(&self) -> AppResult<Event> {
-        Ok(self.receiver.recv()?)
+    /// This function will always block the current thread if there is no data available unless the
+    /// event source has been closed.
+    pub fn next(&self) -> color_eyre::Result<Event> {
+        let event = self.receiver.recv()?;
+        Ok(event)
+    }
+}
+
+/// An event thread that listens for terminal events.
+///
+/// This function is responsible for listening to terminal events and sending them to the main
+/// thread. It sends a tick event at a regular interval. The tick rate is specified by the
+/// `tick_rate` parameter. If an error occurs, it sends an error event to the main thread and then
+/// stops running.
+fn event_thread(sender: mpsc::Sender<Event>, tick_rate: Duration) {
+    let mut last_tick: Option<Instant> = None;
+    loop {
+        if last_tick
+            .map(|tick| tick.elapsed() >= tick_rate)
+            .unwrap_or(true)
+        {
+            sender.send(Event::Tick).expect("failed to send tick event");
```

This should ignore the error. Will fix.